### ISSUE TITLE
Installing XivAlexander makes it impossible for the game to connect to servers

### ISSUE THREAD

**Issue by Marcelonn** [Soreepeong/XivAlexander#161]
I always get the "The lobby server connection has encountered an error" error when trying to log in after having installed XivAlex. Running it without installing works fine.

**Comment by Soreepeong** [comment]
Some things to check:
* Are you using a VPN, "internet accelerator", or some sort of external firewall program? If then, what is it?
* In the log window, when you try to log in with XivAlex installed, does anything get printed? Do you see 127.0.0.1 in log window?
* Does networking work if you launch the game via Loader's "run game" option?
* Does networking work if you "load once" via Loader, and then restart using XivAlexander main menu?

Also, did "Uninstall XivAlexander" button work without hitch?

**Comment by Marcelonn** [comment]
1. No, not using a VPN.
2. This is what gets printed:

> 2021-10-02 08:14:32.452	General	XivAlexander initialized.
> 2021-10-02 08:14:32.495	General	Freed game mutex \BaseNamedObjects\6AA83AB5-BAC4-4a36-9F66-A309770760CB_ffxiv_game00.
> 2021-10-02 08:14:32.495	General	Freed game mutex \BaseNamedObjects\6AA83AB5-BAC4-4a36-9F66-A309770760CB_ffxiv_game_common.
> 2021-10-02 08:14:53.321	SocketHook	15e8: API(socket)
> 2021-10-02 08:14:53.322	SocketHook	15e8: API(connect): 204.2.229.11:54994
> 2021-10-02 08:14:53.846	SocketHook	15e8: Found
> 2021-10-02 08:14:53.846	SocketHook	15e8: Local=192.168.0.8:65284
> 2021-10-02 08:14:53.846	SocketHook	15e8: Remote=204.2.229.11:54994
> 2021-10-02 08:14:53.848	SocketHook	Ping 192.168.0.8 -> 204.2.229.11: Start tracking.

3. If you mean if I can log in by doing that, then yes.
4. Couldn't check that because for some reason I couldn't log in to the official launcher (probably because mine is a steam version of the game).
5. Doesn't work.

Yes, the uninstall function seems to work.

**Comment by Soreepeong** [comment]
[XivAlexander_1_14_8_25_c06689b.zip](https://github.com/Soreepeong/XivAlexander/files/7271731/XivAlexander_1_14_8_25_c06689b.zip)

Can you test out this version? I've made XivAlexander hook socket functions after game main window becomes responsive, which is normally the case when you're using "load once", compared to "install" which makes XivAlexander hook socket functions almost immediately.

**Comment by Marcelonn** [comment]
I tried it and, unfortunately, same error.

**Comment by Soreepeong** [comment]
[XivAlexander_1_14_8_25_TEST2.zip](https://github.com/Soreepeong/XivAlexander/files/7271772/XivAlexander_1_14_8_25_TEST2.zip)

Try this one! It will show you a message like the following:

![image](https://user-images.githubusercontent.com/3614868/135723936-fe1955dd-6618-4cd0-b9db-c06170ac8aaf.png)

See if you can log in without continuing from that button, and when you got to character selection screen, try continuing from the message box and then try logging into a character.

**Comment by Marcelonn** [comment]
I'm really thankful for your attention but it still doesn't work. I did what you said and didn't click "OK" until after I logged in.
![image](https://user-images.githubusercontent.com/80041721/135724213-3ec2cc26-b1af-415e-b42a-9c2bf94a4868.png)

**Comment by Soreepeong** [comment]
I don't really have a clue at this point - I'll get back to you later, when I set up trial account for test with Steam installation. Otherwise, if do have a trial account to use that doesn't involve Steam, try logging in with that one while XivAlex installed. You can launch the game with normal Square Enix account if you launch the game manually via `C:\Program Files (x86)\Steam\steamapps\common\FINAL FANTASY XIV Online\boot\ffxivboot.exe`.

**Comment by dsdamod** [comment]
I seem to have the exact same issue unfortunately.

Only way i can get XivAlexander to work is if i use "load once without instellation" after the game is already running.

**Comment by Pageanator2000** [comment]
I was having the same issue, a clean install seems to have fixed it. 
though windows defender is picking up some of the files as malicious.

**Comment by jekyll87** [comment]
I was having this issue yesterday after updating to the newest version. Did a complete uninstall and also uninstalled Gshade. I was then able to log in to the game and load XivAlexander.

**Comment by ShinjukuCurry** [comment]
Same problem here, installing causes game to not connect. I reverted back to using the portable version.

**Comment by gozoge** [comment]
same problem - i have done some testing and have found:
1: i can launch the game normally and play -
2: once game is launched and i have logged in - if i load xivalexander it will work
3: if i load xivalexander before launch or on launch with the installer - it won't even let me load my game, it gets to the ff14 launcher screen then once hitting play just closes.
4: if i load xiv alexander AND xiv launcher together - i can get the game started, but i get the same error 5006.

**Comment by Nypheena** [comment]
I got this issue as well with version 25

**Comment by Soreepeong** [comment]
I'm pretty sure this happens with Steam installation. If any of you aren't using Steam to play xiv, then let me know which other addons or VPN programs you're using - like GShade, WTFast, and etc.

**Comment by jekyll87** [comment]
I do not use Steam to play XIV. At the time I was having the error, I was using:
GShade
XIVLauncher
BattlePing

I have never had a problem w/ XIVLauncher or Battleping. The uninstallation of XIVAlexander and GShade corrected the 5006 error for me. I re-installed XIVLauncher and worked fine after but I have only tested the "load once" option and not the "install" option after reinstall.

**Comment by Nypheena** [comment]
> I'm pretty sure this happens with Steam installation. If any of you aren't using Steam to play xiv, then let me know which other addons or VPN programs you're using - like GShade, WTFast, and etc.

no steam just xivlauncher integration to count time its mogstation version, protonvpn, IPs excluded. Dalamud plugins, none should be modifying network settings

**Comment by tockity** [comment]
Not using steam and VPN is not running, also running into issues and cannot run the game. The _only_ other program used with XIV is XIVLauncher.

I've reverted back to 1.14.8.19 until a portable version is included again. This whole install business is wholly unnecessary when using xivlauncher.

**Comment by Kamexe** [comment]
Yep I'm having the same issue here when I tried it yesterday. I use XIVlauncher and ACT but nothing else. I get the same error though of cannot connect then force shuts down the game. I had to do uninstall this and now the game works fine again :/

**Comment by Dima6767** [comment]
Having the same issue with the latest version after "installing it"
I'm using Steam version and starting through XIVLauncher.
For now I've resorted to using it without the "installation" part.

**Comment by jamesczhang** [comment]
I have encountered the same issue. XIVLauncher, Astrill VPN are also installed. "Loading once" works fine.

**Comment by Soreepeong** [comment]
[XivAlexander_1_14_8_25_TEST3.zip](https://github.com/Soreepeong/XivAlexander/files/7276673/XivAlexander_1_14_8_25_TEST3.zip)

Can you all test this one? I got a steam sub to test, and in my case, the game wouldn't even start at all, so this one fixes the said problem. I don't know if this is going to fix networking problems as well, but wouldn't hurt to try it out.

**Comment by herrhuml** [comment]
I'm not on steam and I'm using both XIVLauncher and GShade. At first I was crashing, then I couldn't connect to the servers while XivAlexander was installed, loading it once worked fine though. This latest update seems to work just fine even while installed.

**Comment by Marcelonn** [comment]
I'm sorry if this doesn't pertain to the subject of this topic but still I gotta ask: what's the difference between the different install options available in the newest release?

**Comment by Dima6767** [comment]
I'm able to connect to the game now using version 1.14.8.27 and "installing" it.

**Comment by Soreepeong** [comment]
It's literally "try the other one if the first one does not work", in case other addons using the conflicting DLL names do not support XivAlexander's method of resolving it.

**Comment by Soreepeong** [comment]
Additional fixes have been made to not cause problems when the full path to your Documents folder contains non-ASCII characters, and more miscellaneous command line argument stuff that might have been causing login failures.
